frontend/test/pytest/test_pytree_args.py

```diff
@@ -481,7 +481,7 @@ def test_args_used_indirectly_in_measure(self, backend):
         def circuit(dictionary):
             """q0 = 1; q1 = 0;"""
             qml.RX(jnp.pi, wires=0)
-            return measure(dictionary["wire"] + 1 % 2)
+            return measure((dictionary["wire"] + 1) % 2)
```

Comment:
According to [this](https://docs.python.org/3/reference/expressions.html#operator-precedence), `%` comes before `+`, so the added parenthesis actively change the intended behaviour. Taking `1 % 2` doesn't make much sense, so I'm guessing this was just a bug in the test previously?
What prompted the change?

Reply:
Yes, this is an interesting change that I have yet to narrow down. On macOS the runtime would raise an exception saying that a measurement was performed on an invalid wire. This exception is correct according to the operator precedence you link to. It is a bit concerning that:

1. It succeeded in Linux
2. On macOS, without the parenthesis, I believe the failure / success was not deterministic.

I have seen similar behaviour in the past (other compiler passes I was working on) when there was some undefined behaviour and a non-deterministic iteration was performed over some structures in the compiler (all functions in the module for example). But that seems a bit unlikely here.

Reply:
Just noticed that it's even worse, looking at the input we use (`0`) the two cases should be identical (`0 + (1 % 2)` and `(0 + 1) % 2`) 👀 

I think it would be good to investigate further in this case, since there shouldn't be a "qubit" out of bounds error.

Reply:
What do you mean the input we use is 0? Some times it uses 1.

Reply:
```python
        result = circuit({"wire": 0})
        assert jnp.allclose(result, False)
        result = circuit({"wire": 1})
        assert jnp.allclose(result, True)
 ```

Reply:
Oh my bad, must have missed the last bit of code there.

Maybe it is a bug in the runtime? I can confirm that
```py
@qjit
@qml.qnode(qml.device("lightning.qubit", wires=2))
def circuit(x: int):
    return measure(x)
circuit()
```
works for both `0`, `1`, and `2`, but not `3`.

Reply:
I'm not sure yet. I have confirmed that with parenthesis, the MLIR generated actually generates a `remainder` function. Without it, the `remainder` function gets optimized away before MLIR generation. So, the test is more correct after this PR. The question about why it succeeded when it shouldn't, is still open.

Reply:
@dime10 I think this is a bug of its own, so I will just open an issue and resolve this conversation if there is nothing else.

Reply:
The OS thing is really weird, I wonder if there is a difference in the IR generated between the two.

If the order of operations is respected, `remainder` should not be part of the IR since `1 % 2` is just regular Python code that would be transparent to tracing process, unless either argument is a tracer.